server/internal/chatsessions/impl.go

```diff
@@ -0,0 +1,146 @@
+	endpoints.Use(middleware.MapErrors())
+	endpoints.Use(middleware.TraceMethods(service.tracer))
+	srv.Mount(
+		mux,
+		srv.New(endpoints, mux, goahttp.RequestDecoder, goahttp.ResponseEncoder, nil, nil),
+	)
+}
+
+func (s *Service) APIKeyAuth(ctx context.Context, key string, scheme *security.APIKeyScheme) (context.Context, error) {
+	return s.auth.Authorize(ctx, key, scheme)
+}
+
+func (s *Service) ProjectSlugAuth(ctx context.Context, slug string, scheme *security.APIKeyScheme) (context.Context, error) {
+	return s.auth.Authorize(ctx, slug, scheme)
+}
+
+func (s *Service) Create(ctx context.Context, p *gen.CreatePayload) (*gen.CreateResult, error) {
+	ctx, span := s.tracer.Start(ctx, "chatsessions.create")
+	defer span.End()
+
+	authCtx, ok := contextvalues.GetAuthContext(ctx)
+	if !ok || authCtx == nil || authCtx.ActiveOrganizationID == "" {
+		return nil, oops.C(oops.CodeUnauthorized).Log(ctx, s.logger)
+	}
+	if authCtx.ProjectID == nil {
+		return nil, oops.C(oops.CodeUnauthorized).Log(ctx, s.logger)
+	}
+
+	// Validate expires_after (1-3600 seconds)
+	expiresAfter := p.ExpiresAfter
+	if expiresAfter < 1 || expiresAfter > 3600 {
+		return nil, oops.E(oops.CodeBadRequest, nil, "expires_after must be between 1 and 3600 seconds").Log(ctx, s.logger)
+	}
+
+	// Generate JWT token
+	token, _, err := s.chatSessionsManager.GenerateToken(
+		ctx,
+		authCtx.ActiveOrganizationID,
+		authCtx.ProjectID.String(),
+		authCtx.OrganizationSlug,
+		*authCtx.ProjectSlug,
+		p.UserIdentifier,
+		expiresAfter,
+	)
+	if err != nil {
+		return nil, oops.E(oops.CodeUnexpected, err, "failed to generate chat session token").Log(ctx, s.logger)
+	}
+
+	result := &gen.CreateResult{
+		ClientToken:    token,
+		ExpiresAfter:   expiresAfter,
+		Status:         "active",
+		UserIdentifier: p.UserIdentifier,
+	}
+
+	s.logger.InfoContext(ctx, "created chat session",
+		attr.SlogOrganizationID(authCtx.ActiveOrganizationID),
+		attr.SlogProjectID(authCtx.ProjectID.String()),
+	)
+
+	return result, nil
+}
+
+func (s *Service) Revoke(ctx context.Context, p *gen.RevokePayload) error {
+	ctx, span := s.tracer.Start(ctx, "chatsessions.revoke")
+	defer span.End()
+
+	authCtx, ok := contextvalues.GetAuthContext(ctx)
+	if !ok || authCtx == nil || authCtx.ActiveOrganizationID == "" {
+		return oops.C(oops.CodeUnauthorized).Log(ctx, s.logger)
+	}
+	if authCtx.ProjectID == nil {
+		return oops.C(oops.CodeUnauthorized).Log(ctx, s.logger)
+	}
+
+	// Validate the token and extract JTI
+	claims, err := s.chatSessionsManager.ValidateToken(ctx, p.Token)
+	if err != nil {
+		return oops.E(oops.CodeBadRequest, err, "invalid token").Log(ctx, s.logger)
+	}
+
+	// Verify the token belongs to the same org/project
+	if claims.OrgID != authCtx.ActiveOrganizationID || claims.ProjectID != authCtx.ProjectID.String() {
+		return oops.E(oops.CodeForbidden, nil, "token does not belong to this project").Log(ctx, s.logger)
+	}
+
+	// Revoke the token
+	err = s.chatSessionsManager.RevokeToken(ctx, claims.ID)
```

Comment:
not certain about this but wondering if it's better to revoke by the token as the cache key. That way you don't need to parse to check revocation list